tools/src/openssl_signer.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::linux_commands::{execute, execute_and_parse_stdout};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use std::process::Command;
+use sxg_rs::signature::{Format as SignatureFormat, Signer};
+
+#[derive(Debug)]
+pub enum OpensslSigner<'a> {
+    Hmac(&'a [u8]),
+}
+
+#[async_trait(?Send)]
+impl<'a> Signer for OpensslSigner<'a> {
+    async fn sign(&self, message: &[u8], format: SignatureFormat) -> Result<Vec<u8>> {
+        let tmp_file = execute_and_parse_stdout(&mut Command::new("mktemp"))?;
+        let tmp_file = tmp_file.trim();
+        std::fs::write(tmp_file, message)?;
+        match self {
+            OpensslSigner::Hmac(private_key) => {
+                let hexkey = private_key
+                    .iter()
+                    .map(|x| format!("{:02x}", x))
+                    .collect::<Vec<_>>()
+                    .join("");
+                let sig = execute(
+                    Command::new("openssl")
```

Comment:
How do you know openssl is available? Is it part of the installed package?

Reply:
No, we are assuming that the user has installed it, because `openssl` is needed when creating private key.

Reply:
Will there be an error if it's not available?

Reply:
Yes. I just added a new commit, and the error printed to console is now like
```
Error: Failed to sign message

Caused by:
    0: Failed to use openssl to create HMAC
    1: Failed to execute command "openssl" "dgst" "-sha256" "-mac" "HMAC" "-binary" "-macopt" "hexkey:xxxxxx" "/tmp/tmp.xxxxxx"
    2: No such file or directory (os error 2)
```